client/components/data/document-head/index.jsx

```diff
@@ -16,13 +16,26 @@ import {
 	addDocumentHeadMeta as addMeta,
 	setDocumentHeadUnreadCount as setUnreadCount
 } from 'state/document-head/actions';
-import { getSelectedSiteId } from 'state/ui/selectors';
 
 class DocumentHead extends Component {
 	componentWillMount() {
-		this.props.setTitle( this.props.title );
-		this.props.setDescription( this.props.description );
-		this.props.setUnreadCount( this.props.unreadCount );
+		const {
+			title,
+			description,
+			unreadCount
+		} = this.props;
+
+		if ( title ) {
```

Comment:
Will this prevent titles from being reset? I thought there are some cases where we are intentionally setting an empty title, e.g. see #5897.